tests/test_filterLabelFixups.py

```diff
@@ -0,0 +1,118 @@
+    """Tests for the logic in
+    lsst.obs.base.formatters.fitsExposure.FitsExposureFormatter._compareFilters
+    that uses the data ID passed to a formatter to fix and/or check the
+    FilterLabel read from an Exposure FITS file, allowing us to load images
+    with new, standardized filters even if they were written prior to filter
+    standardization (and without enough information to reconstruct the
+    standardized filter name).
+
+    This test lives here instead of obs_base because it relies on having
+    Exposure FITS files written both before and after standardization in a Gen3
+    butler, something trivial to obtain here: the flats are old (from
+    testdata_ci_hsc) - while calexps are new (written by Gen3 pipelines).
+    And this package already has the dependency on a concrete obs package
+    (obs_subaru in this case) necessary to set up a full butler repository,
+    something that obs_base can by definition never have.
+    """
+
+    def setUp(self):
+        self.butler = Butler(os.path.join(getPackageDir("ci_hsc_gen3"), "DATA"), writeable=False,
+                             collections=["HSC/calib/2013-06-17", "HSC/runs/ci_hsc"])
+        # We need to provide a physical_filter value to fully identify a flat,
+        # but this still leaves the band as an implied value that this data ID
+        # doesn't know.
+        self.flatMinimalDataId = DataCoordinate.standardize(
+            instrument="HSC", detector=0, physical_filter="HSC-R",
+            universe=self.butler.registry.dimensions,
+        )
+        self.flatCollection = "HSC/calib/2013-06-17"
+        # For a calexp, the minimal data ID just has exposure and detector,
+        # so both band and physical_filter are implied and not known here.
+        self.calexpMinimalDataId = DataCoordinate.standardize(
+            instrument="HSC", detector=100, visit=903334,
+            universe=self.butler.registry.dimensions,
+        )
+
+    def testReadingOldFileWithIncompleteDataId(self):
+        """If we try to read an old flat with an incomplete data ID, we should
+        get a warning.  It is unspecified what the FilterLabel will have in
+        this case, so we don't check that.
+        """
+        with self.assertWarns(Warning):
+            self.butler.get("flat", self.flatMinimalDataId)
+        with self.assertWarns(Warning):
+            self.butler.get("flat.filterLabel", self.flatMinimalDataId)
+
+    def testFixingReadingOldFile(self):
+        """If we read an old flat with a complete data ID, we fix the
+        FilterLabel.
+        """
+        flatFullDataId = self.butler.registry.expandDataId(self.flatMinimalDataId)
+        flat = self.butler.get("flat", flatFullDataId)
+        self.assertEqual(flat.getFilterLabel().bandLabel, flatFullDataId["band"])
+        self.assertEqual(flat.getFilterLabel().physicalLabel, flatFullDataId["physical_filter"])
+        flatFilterLabel = self.butler.get("flat.filterLabel", flatFullDataId)
```

Comment:
Is there `flat_sub` as well? In gen2 the `_sub` getters needed a separate fix.

Reply:
The syntax for that in Gen3 is to still use `flat` for the dataset type, and add a `parameters={"bbox": ...}` kwarg.  I'm quite confident from the logic in obs_base that that should already work, but it is still a good idea to test it, given that the control flow could of course change.